### DISCUSSION TITLE
search from a subset of indexed documents

### DISCUSSION THREAD

**Discussion by roynirmal** [xhluca/bm25s#81] [General]
Is it possible to index the entire corpus and yet search within a subset of the documents by passing the corresponding doc_ids?

**Comment by xhluca** [comment]
Yes! Please use `weight_mask` in retrieve: https://github.com/xhluca/bm25s/blob/8b5ff105dc550189412da83be68de9a857b325f5/bm25s/__init__.py#L629-L631

The docstring implies only a binary mask is accepted, but you can use any float mask if you want.

**Reply by roynirmal** [reply]
Thanks a lot! Very helpful :)
